### PR TITLE
AI Agents Instructions

### PR DESCRIPTION
**Pull request by tnorling**
This pull request introduces a comprehensive set of agent and contributor instruction documents across the repository, clarifying standards, supported environments, telemetry practices, and changefile requirements. The changes improve onboarding, code quality, and ensure consistent guidance for contributors and AI agents interacting with the codebase.

**Repository-wide agent and contributor instructions:**

* Added a top-level `AGENTS.md` describing the repository structure, package dependencies, build order, and environment setup requirements for the MSAL.js monorepo.
* Introduced a detailed contributor guide in `lib/AGENTS.md` covering code standards, build/validation steps, changefile policies, and strict guidance against using the unsupported instance aware flow.

**Package-specific agent instructions:**

* Added environment support and usage guidelines for each major package:
    - [`msal-common`](diffhunk://#diff-07699d9e1ea0d7c1f010df1ce341c171c7406680cb5fa46ede8380e62f274b15R1-R8): Modern browsers and Node.js 20+
    - [`msal-browser`](diffhunk://#diff-ef24a47ed4dfa2c4e65f1d4454cc29911bfd7e21734bb257e74bec3535b14196R1-R39): Modern browsers, plus telemetry and performance monitoring best practices
    - [`msal-node`](diffhunk://#diff-7193c88f14f8e682e86f5cc1f8bae10fd88b8fd4c4b589adc179f3f08f00e4ffR1-R7): Node.js 20+
    - [`msal-react`](diffhunk://#diff-9ffda21cd4eea6e88236131a08595345473b7d45a969157afa68875ba556a0f5R1-R8): Modern browsers and React 19+
    - [`msal-angular`](diffhunk://#diff-b9dcc1b4dbef77e982f53b86d5924882c2ef422d465c1e41e5dc697a4c339cc6R1-R8): Modern browsers and Angular 19+

**Sample and test instructions:**

* Added `samples/AGENTS.md` to clarify sample usage, best practices, and reinforce the prohibition on deprecated features and instance aware flow in sample code.

**Changefile process documentation:**

* Created `.github/instructions/changefiles.instructions.md` to standardize changefile types, messages, and review practices for all contributors.

> [!NOTE]
> Github documents in what contexts each instruction file is used [here](https://docs.github.com/en/copilot/concepts/prompting/response-customization?tool=webui#support-for-repository-custom-instructions). Most of the commonly used AI Agents now support the `AGENTS.md` files as a standard

### CODE REVIEW

File: lib/msal-react/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# msal-react Instructions
+
+## Supported Environments
+
+- msal-react supports all modern mainstream browsers (Chrome, Firefox, Safari, Edge)
+- msal-react supports React versions 19+
```
[the remark this answers is not in the document]
Comment (tnorling) [reply lib/msal-react/AGENTS.md:6]: This branch is specifically v5, older versions are on other branches

File: lib/msal-angular/AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# msal-angular Instructions
+
+## Supported Environments
+
+- msal-angular supports all modern mainstream browsers (Chrome, Firefox, Safari, Edge)
+- msal-angular supports Angular versions 19+
```
[the remark this answers is not in the document]
Comment (tnorling) [reply lib/msal-angular/AGENTS.md:6]: This branch is specifically v5, older versions are on other branches

**Review by peterzenz — APPROVED** [review]
So separate folder level "Agent" instructions is all we need here?

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -4,466 +4,41 @@
...
-### Code Suggestion Best Practices
+When reviewing pull requests, GitHub Copilot should provide comprehensive feedback focusing on four key areas:
 
-1. **Be specific**: Each suggestion should target a specific issue
-2. **Include context**: Briefly explain why the change is needed
-3. **Keep it simple**: Break complex changes into multiple suggestions
-4. **Link documentation**: Reference relevant docs or standards when applicable
-5. **Suggest liberally**: Code suggestions are easy to accept or decline
+1. Suggest documentation updates for new public methods, properties and APIs, changes to existing APIs, new error scenarios or codes, performance considerations, breaking changes, and usage examples
+1. Suggest adding test coverage (if not included) for new functions, properties, error and edge cases. Complex features should include E2E tests.
```
Comment (sameerag) [inline .github/copilot-instructions.md:42]: Should we introduce a tag for "complex features"?

File: .github/instructions/changefiles.instructions.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+applyTo: "**/change/**"
+---
+
+# Changefiles
+
+Change type should always be either `patch` or `minor` for any changes to files in the `src` directory or changes that may impact the build output. Pull requests that only make changes to `docs` and/or `test` should always have a change type of `none`.
```
Comment (sameerag) [inline .github/instructions/changefiles.instructions.md:7]: Eventually can we flag API changes that will need "major" instead of copilot/cline auto qualifying to suggested categories only?

File: lib/AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+Before committing changes to any library's source code ensure all of the following steps have been completed without errors for each of the affected packages.
+All of the following commands should be run in each package directory where changes were made.
+
+1. Run `npm run build:all` to ensure the code builds without errors
+1. Run `npm run lint` to ensure the code adheres to the coding standards
+1. Run `npm test` to ensure all tests pass
+1. Run `npm run format:check` to ensure the code is formatted correctly
+    - If there are formatting errors, run `npm run format:fix` to automatically fix them
+1. Run `npm run apiExtractor` to ensure the API documentation is up to date
+    - If there are changes to the API documentation, run `npm run apiExtractor -- --local` to update the API documentation
```
Comment (sameerag) [inline lib/AGENTS.md:28]: We should add `npm audit` at some point and auto fix dependencies that can be fixed. Will track it separately, just adding for traction.

**Review by sameerag — APPROVED** [review]
Added some comments, agree it is iterative. Approving.

File: lib/AGENTS.md (hunk 5 above)
Comment (tnorling) [reply lib/AGENTS.md:28]: Generally this should be handled by dependabot rather than Copilot. We can see if there's things we can do to reduce the friction there but I don't expect this is something we'll add here

File: .github/instructions/changefiles.instructions.md (hunk 4 above)
Comment (tnorling) [reply .github/instructions/changefiles.instructions.md:7]: Good callout we can workshop it for the next major
